### PR TITLE
feature: S3C-2773 object lock design document

### PR DESCRIPTION
**Pull request by rahulreddy**

### CODE REVIEW

File: docs/design/object-lock.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+SEC 17a-4 compliance.
+
+## Requirements
+
+* Object lock flag must be set during bucket creation
+* Versioning has to be enabled on the bucket
+* Object lock must be enabled on a bucket in order to write a lock configuration
+  using the PUT Object Lock Configuration api that has object lock flag set
+
+PS: AWS specification does not have a way of setting object lock on existing
```
Comment (dora-korpar) [inline docs/design/object-lock.md:16]: It looks like you wrote 2 different sentences here in one

File: docs/design/object-lock.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+## Deviations
+
+The following hightlights any deviation from AWS implementation of the feature
+
+-  `token`
+
+  AWS allows setting object lock on existing buckets by asking the user to
+  contact support, the support then gives a `token` that can be used to set lock
+  on the specific bucket. Cloudserver's implementation differs in the way that
```
Comment (dora-korpar) [inline docs/design/object-lock.md:186]:
```suggestion
  on the specific bucket. Cloudserver's implementation differs in that
```

File: docs/design/object-lock.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+* **Get Object Legal Hold** - gets the legal hold status for an object version
+
+## Deviations
+
+The following hightlights any deviation from AWS implementation of the feature
+
+-  `token`
+
+  AWS allows setting object lock on existing buckets by asking the user to
+  contact support, the support then gives a `token` that can be used to set lock
```
Comment (dora-korpar) [inline docs/design/object-lock.md:185]:
```suggestion
  contact support. The support then gives a `token` that can be used to set lock
```

File: docs/design/object-lock.md (hunk 1 above)
Comment (rahulreddy) [reply docs/design/object-lock.md:16]: That's true, not sure what happened there. I think I used Github's commit suggestion in the prior PR and it did not do things right

**Pull request merged**